### PR TITLE
`ToggleGroupControl`: remove animated backdrop

### LINKED ISSUE #37506: Color select component backdrop render and animation opportunities
### Description

The rendering on the backdrop is inconsistent when rendering inside an animating parent. The current implementation has a delay of 100ms but it's not very reliable, and is also dependent on the parent animating for 100ms.

### Step-by-step reproduction instructions

1. Add a group block
2. Open the color settings
3. Press Background color
4. Toggle gradient and solid items

### Screenshots, screen recording, code snippet

https://user-images.githubusercontent.com/1478421/146626852-1d982150-080d-48bb-980d-08968087a887.mov

### Environment info

- Gutenberg 12.2.0-rc.1
- TT1 Blocks
- Chrome
- Mac

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by KevinBatdorf**
## Description

This PR removes the backdrop animation component and replaces it with a simple background color change

This PR is an alternative solution to #37507 

Fixes #37506

## How has this been tested?

Tested by adding a component with a background color and swapping between gradient and solid. Also, when the container is focused, pressing the left and right arrow keys properly applies the background color.

## Screenshots 

https://user-images.githubusercontent.com/1478421/149639228-9f3b2caf-50bf-49a4-b6f7-ab531aacdaa1.mov

## Types of changes

Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [ ] I've updated related schemas if appropriate.

### CODE REVIEW

File: packages/components/src/toggle-group-control/toggle-group-control/component.tsx [hunk 1]
```diff
@@ -49,7 +48,7 @@ function ToggleGroupControl(
 	} = useContextSystem( props, 'ToggleGroupControl' );
 	const cx = useCx();
 	const containerRef = useRef();
-	const [ resizeListener, sizes ] = useResizeAware();
+	const [ resizeListener ] = useResizeAware();
```
Comment (ciampo) [inline packages/components/src/toggle-group-control/toggle-group-control/component.tsx]: It looks like `react-resize-aware` was only needed to resize the backdrop, and so I think we can remove this line and the import at the top of this file (cc'ing @ntsekouras for a confirmation!)

**Review by ciampo — COMMENTED** [review]
Thank you @KevinBatdorf for opening this PR!

The code changes test well, and overall LGTM. Would you be able to add an entry to the [CHANGELOG](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CHANGELOG.md)?

The e2e failures don't seem related, maybe rebasing could be enough to have them pass.

File: packages/components/src/toggle-group-control/toggle-group-control/component.tsx (hunk 1 above)
Comment (KevinBatdorf) [reply packages/components/src/toggle-group-control/toggle-group-control/component.tsx]: That makes sense actually. The listener isn't very useful if the `sizes` aren't being used. I removed it for now

**Comment by KevinBatdorf** [comment]
Hey @ciampo I added the changelog and removed the listener. I also had to update the test to remove the listener too. Some of the tests are still failing, but I don't think they are related to this change?

File: packages/components/src/toggle-group-control/toggle-group-control/component.tsx (hunk 1 above)
Comment (ntsekouras) [reply packages/components/src/toggle-group-control/toggle-group-control/component.tsx]: Yeah, it seems it can be safely removed.

**Review by ntsekouras — APPROVED** [review]
Thanks for your work and all the iterations @KevinBatdorf! 👏 

LGTM and I agree with @ciampo about this being  a better short-term solution for the issue.
With a green CI let's 🚢

**Review by ciampo — APPROVED** [review]
Thank you Kevin, LGTM 🚀

**Comment by jasmussen** [comment]
Is there anything we can do to follow up and get the animated backdrop back in the next few days? Although I appreciate the need to fix separate bugs, this is a significant design regression.

**Comment by ciampo** [comment]
> Is there anything we can do to follow up and get the animated backdrop back in the next few days? Although I appreciate the need to fix separate bugs, this is a significant design regression.

Implementing this transition effectively and in a bug-free manner can prove quite complicated (see the complexity of the code changes in the last bug fix attempt in #37507), and I can't guarantee that I'll be able to come up with a solution in just a few days.

If, for the time being, you'd rather keep the older animation and reintroduce the bug that this PR fixed, I can go ahead an revert this PR. I will then work separately on a new implementation for the animation, and merge it when it's ready.

**Comment by jasmussen** [comment]
We'll want to fix the bug, of course, but I'd prefer keeping the bug, and the backdrop until we can get both working at the same time. Thanks all for the work!

**Pull request merged**